docs/static/doc-for-plugin.asciidoc

```diff
@@ -0,0 +1,184 @@
+[[plugin-doc]]
+=== Documenting your plugin
+
+Quality documentation with good examples contributes to the adoption of your plugin.
+
+The documentation that you write for your plugin will be generated and published
+in the {logstash-ref}/index.html[Logstash Reference] and the
+{lsplugindocs}[Logstash Versioned Plugin Reference].
+
+[[plugin-doc-file]]
+==== Documentation file
+
+Documentation is a required component of your plugin.
+It belongs in a single file called 'docs/index.asciidoc'.
+The <<plugin-generator,plugin generation utility>> creates a starter file for you.  
+
+[[heading-ids]]
+==== Heading IDs
+
+Format heading anchors with variables that can support generated IDs. This approach
+creates unique IDs when the {lsplugindocs}[Logstash Versioned Plugin Reference]
+is built. Unique heading IDs are required to avoid duplication over multiple versions of a plugin.
+
+*Example*
+
+Don't hardcode a heading ID like this: `[[config_models]]`
+
+Instead, use variables to define it:
+
+[source,txt]
+----------------------------------
+[id="plugins-{type}s-{plugin}-config_models"]
+==== Configuration models
+----------------------------------
+
+If you hardcode an ID, the {lsplugindocs}[Logstash Versioned Plugin Reference]
+builds correctly the first time. The second time the doc build runs, the ID
+is flagged as a duplicate, and the build fails.
+
+
+[[link-format]]
+==== Link formats
+
+Correct link formatting is essential for directing users to the content you
+want them to see. Incorrect link formatting or duplicate links can break the
+documentation build. Let's not do that. 
+
+===== Link to content in the same file
+
+Use angle brackets to format links to content in the same asciidoc file. 
+
+*Example*
+
+This link:
+[source,txt]
+-----
+<<plugins-{type}s-{plugin}-config_models>>
+-----
+
+Points to this heading in the same file:
+
+[source,txt]
+----------------------------------
+[id="plugins-{type}s-{plugin}-config_models"]
+==== Configuration models
+----------------------------------
+
+
+===== Link to content in the Logstash Reference Guide
+
+Use external link syntax for links that point to documentation for other plugins or content in the Logstash Reference Guide.
+
+*Examples*
+[source,txt]
+-----
+{logstash-ref}/plugins-codecs-multiline.html[Multiline codec plugin]
+-----
+
+[source,txt]
+-----
+{logstash-ref}/getting-started-with-logstash.html
+-----
+
+===== Link text
+
+If you don't specify link text, the URL is used as the link text.
+
+
+*Examples*
+
+If you want your link to display as {logstash-ref}/getting-started-with-logstash.html, use this format:
```

Comment:
Do we want users to do this though? (Just asking...I don't have a strong feeling, though it's not typical for us to show the full URL in doc links.)